snippets/csharp/System.Reflection/EventInfo/AddEventHandler/source.cs

```diff
@@ -28,11 +28,11 @@ public static void Main()
         // creates an array of Type objects that represent the
         // parameter types of the Invoke method.
         //
-        Type handlerType = eInfo.EventHandlerType;
-        MethodInfo invokeMethod = handlerType.GetMethod("Invoke");
-        ParameterInfo[] parms = invokeMethod.GetParameters();
-        Type[] parmTypes = new Type[parms.Length];
-        for (int i = 0; i < parms.Length; i++)
+        Type? handlerType = eInfo?.EventHandlerType;
+        MethodInfo? invokeMethod = handlerType?.GetMethod("Invoke");
+        ParameterInfo[]? parms = invokeMethod?.GetParameters();
+        Type[] parmTypes = new Type[parms is not null ? parms.Length : 0];
```

Comment:
this might be slightly simpler:

```suggestion
        Type[] parmTypes = new Type[parms?.Length ?? 0];
```